### ISSUE TITLE
Disable autoplay tracking ticket

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#8739]
**Describe the issue you encountered:**

- [x] This video should play when you press the play button regardless of how autoplay is set for this page. https://github.com/brave/browser-laptop/issues/8739#issuecomment-302284448 #8897

When autoplay disabled:
- [x] Let users understand that autoplay is disabled by default by adding layer/placeholder (like flash object) for example
    - enabling autoplay on the placeholder should not reload the whole page
- [ ] Stop downloading of video (saving data consumption is one of the main reasons to disable autoplay)
    - If it is not possible, do not reload the page when the switch is changed. It just consumes the data double.

Also:
- [x] Change "Allow Autoplay Media" with disabled by default into "**Block Autoplay Media**" with enabled by default (this is a suggestion from a viewpoint of switch consistency and UX; on the other switches the state "on" means "block", while on the autoplay switch the state "on" means "unblock")

----

- Platform (Win7, 8, 10? macOS? Linux distro?): Debian

- Brave Version (revision SHA): master branch

- Steps to reproduce:
    1. Open any youtube video
    2. Click the top right lion icon

- Extra QA steps:
    1.
    2.
    3.

- Any related issues: #8738 #8847 #2227

**Comment by luixxiul** [comment]
If those issues could block the release, please change the milestone or add labels. thanks!

**Comment by jonathansampson** [comment]
@darkdh @bsclifton Thoughts on making this higher-priority/release-blocking given its impact on YouTube?

**Comment by darkdh** [comment]
fixed and added blocking label

**Comment by darkdh** [comment]
About `Stop downloading of video`, you can see it actually stops downloading when paused(being autoplay blocked)

**Comment by luixxiul** [comment]
https://github.com/brave/browser-laptop/pull/8751#pullrequestreview-36875944 per @cezaraugusto 

> Testing on YT, video kept loading until I decided what to do with notification. Deciding to disable autoplay, loading persists and pause button is shown rather than play button. Seems like video is not fully loaded and I had to click twice on the pause button to get it working. I felt confused by that.
> 
> Regarding UX the best scenario I can think of is having video stopped (play indicator visible) with video placeholder showing instead of forever loading until user decide what to do with notification.

**Comment by srirambv** [comment]
I would vote for how flash blocker works on Chrome. It just blocks the content and when clicked on run just let only the flash content run  and reload the page. As a user its really annoying if the page reloads everytime the blocked content is loaded especially for videos.

**Comment by jonathansampson** [comment]
@srirambv That would be idea, but would presumably require some significant re-engineering.

**Comment by luixxiul** [comment]
I thought autoplay was enabled by default. 

If not, I think we should find a way of changing the autoplay global settubg easily (eg. add a link/button on the notificaiton) as this was a radical change. 

It must be hard for ordinary users to find out how to change the setting (I believe they cannot imagine that the setting is under `Security` tab). CC @srirambv for his opinion on this.

**Comment by luixxiul** [comment]
Adding another notification bar would work, such as:

<img width="985" alt="screenshot 2017-05-17 14 58 15" src="https://cloud.githubusercontent.com/assets/3362943/26140390/7333930e-3b11-11e7-84ce-40cc977e12a0.png">

Imagine there are 3 buttons on the 1st bar, `Change global setting`(-> Open about:preferences#security), `Yes`, and `No`.

**Comment by srirambv** [comment]
>It must be hard for ordinary users to find out how to change the setting (I believe they cannot imagine that the setting is under Security tab). 

Agree users will definitely find it hard to search for the setting. I would suggest add this to the welcome page that @cezaraugusto has worked on.  This way once the welcome page is implemented and shown, it would also show where the user can set the setting. 

A generic notification bar would be better with a button to link to settings (similar to ledger notification bar) 

![image](https://cloud.githubusercontent.com/assets/17010094/26143375/9b938488-3adc-11e7-8fd0-a9663a7c5561.png)

**Comment by jonathansampson** [comment]
@darkdh @bradleyrichter Your thoughts on this?

**Comment by bradleyrichter** [comment]
This is unfortunately broken.

go here: https://blog.google/products/photos/google-photos-500-million-new-sharing/

Then try to play the main video. It spins until you accept the auto-play alert bar. 

This is not how it should work. This video should play when you press the play button regardless of how autoplay is set for this page.

Must fix or pull.  :  (

**Comment by luixxiul** [comment]
c5f44099acdc56ec09e56c448df9f7e76b3c37f6

**Comment by Jacalz** [comment]
Yeah the same is for svt.se @bradleyrichter

**Comment by Jacalz** [comment]
Spotify needs to be disabled :D

**Comment by bsclifton** [comment]
from @bradleyrichter, regarding https://boingboing.net/2017/06/21/brave-orfox-firefox-focus.html
> We need to improve the auto-play blocking. The top video on this page does not seem to be set to autoplay but we make it hard for the user to play the video which won’t play until you accept the alert.

**Comment by bsclifton** [comment]
Closing as stale

There are open issues tracking autoplay functionality in [brave-core](https://github.com/brave/brave-browser):
https://github.com/brave/brave-browser/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+autoplay
